packages/http/src/validator/index.ts

```diff
@@ -68,29 +68,50 @@ const tryValidateInputBody = (
   requestBody: IHttpOperationRequestBody,
   bundle: unknown,
   body: unknown,
-  mediaType: string
+  headers: IHttpNameValue
 ) =>
   pipe(
     checkBodyIsProvided(requestBody, body),
     E.chain(() => {
+      const headersNormalized = caseless(headers || {});
+
+      const contentLength = parseInt(headersNormalized.get('content-length')) || 0;
+      if (contentLength === 0) {
+        // generously allow this content type if there isn't a body actually provided
+        return E.right(body);
+      }
+
+      const mediaType = headersNormalized.get('content-type');
```

Comment:
Now that I've seen how you did this, I like that it's inside `validateInputIfBodySpecIsProvided()`.

Reply:
samesies! 🙌  Nice work!